### PR TITLE
add resource request/limit to metrics collector pod

### PR DESCRIPTION
**Pull request by haibinxie**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: drivers/storage/portworx/component/telemetry.go [hunk 1]
```diff
@@ -281,6 +289,16 @@ func (t *telemetry) getCollectorDeployment(
 						{
 							Name:  "collector",
 							Image: collectorImage,
+							Resources: v1.ResourceRequirements{
+								Requests: v1.ResourceList{
+									v1.ResourceMemory: resource.MustParse(defaultCollectorMemoryRequest),
+									v1.ResourceCPU:    cpuQuantity,
+								},
+								Limits: v1.ResourceList{
```
Comment (pure-jliao) [inline drivers/storage/portworx/component/telemetry.go:297]: don't need cpu limit here?

File: drivers/storage/portworx/component/telemetry.go [hunk 2]
```diff
@@ -281,6 +289,16 @@ func (t *telemetry) getCollectorDeployment(
 						{
 							Name:  "collector",
 							Image: collectorImage,
+							Resources: v1.ResourceRequirements{
+								Requests: v1.ResourceList{
+									v1.ResourceMemory: resource.MustParse(defaultCollectorMemoryRequest),
+									v1.ResourceCPU:    cpuQuantity,
```
Comment (pure-jliao) [inline drivers/storage/portworx/component/telemetry.go:295]: could use `resource.MustParse(defaultCollectorCPU)` as well

**Review by pure-jliao — APPROVED** [review]

File: drivers/storage/portworx/components_test.go [hunk 3]
```diff
@@ -11455,6 +11455,12 @@ func TestTelemetryEnableAndDisable(t *testing.T) {
 	require.NoError(t, err)
 	require.Len(t, deployment.OwnerReferences, 1)
 	require.Equal(t, cluster.Name, deployment.OwnerReferences[0].Name)
+	memReq := deployment.Spec.Template.Spec.Containers[0].Resources.Requests[v1.ResourceMemory]
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/components_test.go:11458]: Why don't we compare the deployment against a testspec like we do in other components?

**Review by piyush-nimbalkar — APPROVED** [review]

File: drivers/storage/portworx/component/telemetry.go (hunk 1 above)
Comment (haibinxie) [reply drivers/storage/portworx/component/telemetry.go:297]: probably no customer requested it, other components also do not have it.

File: drivers/storage/portworx/components_test.go (hunk 3 above)
Comment (haibinxie) [reply drivers/storage/portworx/components_test.go]: sounds good, resolved in another commit.

**Pull request merged**